### DISCUSSION TITLE
Max number of relationship found in relational picker component

### DISCUSSION THREAD

**Discussion by TommasoBernini** [Budibase/budibase#7183] [Help] [answered]
Hi everyone,
following my previous [question](https://github.com/Budibase/budibase/discussions/7167) I've successfully created 3+ relationships between the table I'm working on, and the other linked tables in my database. 
In the update form for the table in consideration I'm using 3+ relational picker component, however, it seemed that only 3 relationships have been correctly stored. I've also tried to delete and recreate the missing relations, the result is that the newly created relations are found by the relational picker, but the old ones are lost, in such a way that only up to 3 relationships can be used by the relational picker in the same form. It's a bug or I'm missing something?

Here are the relations are shown in the dropdown menu  
![Schermata 2022-08-09 alle 17 49 35](https://user-images.githubusercontent.com/24413162/183703743-444275e2-40eb-4b6e-ad29-8063890ba16f.png)

here the missing ones
![Schermata 2022-08-09 alle 18 15 33](https://user-images.githubusercontent.com/24413162/183704319-a002d7d7-2a35-43e0-bbdf-34607d5d5063.png)

Thank you in advance for your help :)

**Comment by R2bEEaton** [comment]
Hey @TommasoBernini! Can you send an export of your app so I can examine this further?

**Reply by TommasoBernini** [reply]
Unfortunately not, there's our database linked to the app, however, I can guide you into the process to replicate the error:

- Create 4 or plus tables where each row is identified by a UUID and, for example, Table 1 is made like this:

   Columns: UUID_table_1, UUID_table_2, UUiD_table_3, UUiD_table_4, UUiD_table_5, some other data
  where at least 4 or plus columns recall another table row.

- Create a relation in each linked table in Table 1 through the UUID_table_n, the relation must be more than 3;
- Create an update form for the rows in Table 1 where the data of the linked columns are set by a relational picker;
- The problem emerges now, the lists of available fields (the linked columns) in the relational picker component do not go past 3 available relations despite the existence of more relations.

Let me know if the steps above are clear or if you need anything else!

**Reply by Landau-ux** [reply]
It would be super usefull!

**Comment by melohagan** [comment, marked as answer]
Hi @TommasoBernini 

I appreciate that you cannot share your data, but I would need to see the table schemas to accurately recreate this issue. 
I want to be sure that your primary and foreign keys are setup correctly.
I would also want to know how you have defined the relationships, e.g. one->many, many->many, and which ids were used as primary, foreign, as well as any through tables.

**Reply by TommasoBernini** [reply]
The problem fixed itself by deleting and re-assigning all of the relationships

**Reply by melohagan** [reply]
Glad it works now. If you ever find the issue pops up again, it would be good if you can create a bug with the reproduction steps. Thanks for letting me know that you got it working!

**Comment by diogenesbrussels** [comment]
Hi,

I think I have the same problem:
I have two SQL tables with a one-to-many relationship (a citizen and the city he comes from), the drop-down menu contains only 15 cities out of the 24 present in the table.
How can I identify the problem and solve it?

Thank you!

**Reply by diogenesbrussels** [reply]
Some additional information:
The request https://**/api/datasource_plus_**__cities/rows for the relationship dropdown
- In fact, the query returns 16 results classified by ascending id order
- The query also returns all linked rows for each city, with more than 6,000 citizens, this makes the response heavy without being necessary?

**Reply by TommasoBernini** [reply]
I now have again a very similar problem. Two tables, with a one-to-many relationship between them (table A are sectors and B prices with the sector propriety), the relational picker that refers to the sector of a specific price, shows only two sectors out of 15. Moreover, despite the rows of the A table being present in the DB (Postgres), when I look and the database's tables in Budibase data, only a subset of the rows in the sector table are shown. Now, when I order the rows using the A-Z sorting of a specific column, makes some of the missing rows pop up. (app version: 2.0.18)

**Reply by diogenesbrussels** [reply]
Strange behaviour, I have all the rows in the Budibase table view 🤷‍♂️

**Reply by diogenesbrussels** [reply]
Workaround: remove the relationship picker field, create a custom query to get the list, add a data provider that uses this query, add an option picker with the data provider, force the name to match the relationship field (fk_...)

I have already done this for 6 different fields, it complicates the creation of the forms but the relationship picker doesn't seem to be reliable (or rather the access to all rows via the API)
